pupil-spa/src/app/question.service.ts

```diff
@@ -55,4 +54,22 @@ export class QuestionService {
 
     return currentQuestionNumber + 1;
   }
+
+  getConfig() {
+    return this.config;
+  }
+
+  reset() {
+    this.questions = null;
+    this.config = null;
+  }
+
+  initialise() {
```

Comment:
is there any reason why you aren't put the contents of `initialise()` in the constructor?  It would save you from having to call it in every test.